server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -204,14 +234,15 @@ public float getShardBalance() {
      * </ul>
      * <code>weight(node, index) = weight<sub>index</sub>(node, index) + weight<sub>node</sub>(node, index)</code>
      */
-    private static class WeightFunction {
+    public static class WeightFunction {
```

Comment:
lets keep the class private along with its constructor or any method which has been changed to public

Reply:
I think this was needed for tests - so we can have more scoped UTs that test the weight fn alone. Existing balancer tests always tested the full reroute path. We can explore if package private helps.